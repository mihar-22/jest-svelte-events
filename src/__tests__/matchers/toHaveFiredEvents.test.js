import '../../'
import { render, fireEvent } from '@testing-library/svelte'
import Comp from '../fixtures/Comp.svelte'

describe('.toHaveFiredEvents', () => {
  test('passes when given events that fired', async () => {
    const { component, getByText } = render(Comp)
    listen(component, ['e1', 'e2'])
    const e1Button = getByText('e1')
    await fireEvent.click(e1Button)
    const e2Button = getByText('e2')
    await fireEvent.click(e2Button)
    expect(component).toHaveFiredEvents(['e1', 'e2'])
  })

  test('fails when given events that did not fire', () => {
    const { component } = render(Comp)
    listen(component, ['e1', 'e2'])
    expect(() => {
      expect(component).toHaveFiredEvents(['e1', 'e2'])
    }).toThrowErrorMatchingSnapshot()
  })
})

describe('.not.toHaveFiredEvents', () => {
  test('passes when given events that did not fire', () => {
    const { component } = render(Comp)
    listen(component, ['e1', 'e2'])
    expect(component).not.toHaveFiredEvents(['e1', 'e2'])
  })

  test('fails when given events that did fire', async () => {
    const { component, getByText } = render(Comp)
    listen(component, ['e1', 'e2'])
    const e1Button = getByText('e1')
    await fireEvent.click(e1Button)
    expect(() => {
      expect(component).not.toHaveFiredEvents(['e1', 'e2'])
    }).toThrowErrorMatchingSnapshot()
  })
})
