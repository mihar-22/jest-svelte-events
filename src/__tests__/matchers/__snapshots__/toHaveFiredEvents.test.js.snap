// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toHaveFiredEvents fails when given events that did fire 1`] = `
"[2mexpect([22m[31mcomponent[39m[2m).[22mnot[2m.[22mtoHaveFiredEvents[2m([22m[32mexpected[39m[2m)[22m

Expected not to have fired: [32m[\\"e1\\", \\"e2\\"][39m
Recevied events: [31m[\\"e1\\"][39m"
`;

exports[`.toHaveFiredEvents fails when given events that did not fire 1`] = `
"[2mexpect([22m[31mcomponent[39m[2m).[22mtoHaveFiredEvents[2m([22m[32mexpected[39m[2m)[22m

Expected to have fired: [32m[\\"e1\\", \\"e2\\"][39m
Recevied events: [31m[][39m"
`;
